spec/associations_spec.cr

```diff
@@ -55,6 +68,18 @@ describe Avram::Model do
       post2.tags.size.should eq 0
     end
 
+    it "joins associations with the query option" do
+      tag = TagBox.create
+      funny_tag = TagBox.create &.name(Tag::FUNNY_TAG_NAME)
+      post = PostBox.create
+
+      TaggingBox.new.tag_id(tag.id).post_id(post.id).create
+      TaggingBox.new.tag_id(funny_tag.id).post_id(post.id).create
+
+      post.tags.should eq [tag, funny_tag]
+      post.funny_tags.should eq [funny_tag]
```

Comment:
This spec fails `Nil assertion failed (NilAssertionError)` which wasn't very helpful running this locally... Maybe @matthewmcgarvey has some idea?